### PR TITLE
Deploy all releases in parallel

### PR DESCRIPTION
**Pull request by dasch**
This will totally work.

/cc @zendesk/samson

### CODE REVIEW

**Comment by grosser** [comment]
we did some work to make less deploys start in parallel, so this would kinda undo that ... also it should not take that long to start the deploys, they are just put into the queue

/cc @ragurney

**Comment by dasch** [comment]
Because of Kubernetes API server load? We currently spend 2.5 minutes sending out the release docs, and this will only grow as we add more pods (6 more coming).

How many can safely be sent in parallel to a single API server?

**Comment by grosser** [comment]
ah this is inside the deploy-executor ... that should be fine

File: plugins/kubernetes/app/models/kubernetes/deploy_executor.rb [hunk 1]
```diff
@@ -372,14 +372,17 @@ def grouped_deploy_group_roles
...
-      release_docs.each do |release_doc|
+      release_docs.map do |release_doc|
         puts_action "Deploying", release_doc
-        if release_doc.blue_green_color
-          non_service_resources(release_doc).each(&:deploy)
-        else
-          release_doc.deploy
+
+        Thread.new do
+          if release_doc.blue_green_color
```
Comment (grosser) [inline plugins/kubernetes/app/models/kubernetes/deploy_executor.rb]: prefer to map first and then threaded `.deploy` to separate concerns a bit better

Comment (grosser) [reply plugins/kubernetes/app/models/kubernetes/deploy_executor.rb]: also use `Parallel.map` so exceptions are handled (we don't need to DB multithreading that Samson::Parallelizer provides)

**Comment by dasch** [comment]
@grosser I don't have time to get Samson local development set up – will you be able to finish this once Bootstrapgate is finished?

**Comment by dasch** [comment]
Thanks!

**Comment by grosser** [comment]
```
before:
[20:12:47] Deploying Pod99 role worker
[20:12:52] Deploying Pod99 role console
[20:12:53] Deploying Pod98 role worker
[20:12:57] Deploying Pod98 role console

after:
[18:51:01] Deploying Pod99 role worker
[18:51:01] Deploying Pod99 role console
[18:51:01] Deploying Pod98 role worker
[18:51:01] Deploying Pod98 role console

```

**Comment by dasch** [comment]
❤️❤️❤️

**Pull request merged**